### PR TITLE
[BEAM-6184]Make checkstyle report error on missing javaDocMethod

### PR DESCRIPTION
**Pull request by HuangLED**
1. Turn on javadocmethod checkstyle (i.e. now reports error)
2. Added suppressions to all the violations. 

Also created another beam-6446 to track the future progress of removing suppressions. 

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Seed Job

**Comment by asfgit** [comment]
SUCCESS 
 
--none--

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by HuangLED** [comment]
Run Python PreCommit

**Comment by HuangLED** [comment]
R: @robertwb  @kennknowles 

CC: @swegner

File: sdks/java/build-tools/src/main/resources/beam/suppressions.xml [hunk 1]
```diff
@@ -30,4 +30,65 @@
   <suppress checks=".*" files=".+[\\\/]generated-src[\\\/].+\.java" />
   <suppress checks=".*" files=".+[\\\/]generated-sources[\\\/].+\.java" />
   <suppress checks=".*" files=".+[\\\/]generated-test-sources[\\\/].+\.java" />
+
+  <!-- Suppression on JavadocMethod violations.
```
[the remark this answers is not in the document]
Comment (HuangLED) [reply sdks/java/build-tools/src/main/resources/beam/suppressions.xml]: No, with some research I didn't find an automated way. Had to manually create this list. 

My plan is to periodically come back to this, try removing them and see which still leads to error.  Created a tracking JIRA as a long term reminder.

File: sdks/java/build-tools/src/main/resources/beam/suppressions.xml [hunk 2]
```diff
@@ -30,4 +30,65 @@
...
+  <suppress checks="JavadocMethod" files=".*sdk.*DoFn\.java" lines="444"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*CoGbkResult\.java" lines="72"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*extensions.*Audience\.java" lines="32"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*clickhouse.*ClickHouseIO\.java" lines="467"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*clickhouse.*ClickHouseWriter\.java" lines="47"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*sql.*BeamSetOperatorsTransforms\.java" lines="59"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*nexmark.*GeneratorConfig\.java" lines="89"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*CassandraServiceImpl\.java" lines="112"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*gcp.*AvroUtils\.java" lines="28"/>
+  <suppress checks="JavadocMethod" files=".*sdk.*testing.*BigqueryClient\.java" lines="277,357,454"/>
```
[the remark this answers is not in the document]
Comment (HuangLED) [reply sdks/java/build-tools/src/main/resources/beam/suppressions.xml]: For conciseness. 

The intention is indeed to have the full file name (in this particular case BigqueryClient.java is a full file name).

[the remark this answers is not in the document]
Comment (swegner) [reply sdks/java/build-tools/src/main/resources/beam/suppressions.xml]: Got it. So the 3 numbers here refer to different method lengths?

[the remark this answers is not in the document]
Comment (HuangLED) [reply sdks/java/build-tools/src/main/resources/beam/suppressions.xml]: 3 numbers refer to 3 functions (in the same file) that all violate our check, each number for one distinct function.

**Comment by HuangLED** [comment]
Also, this PR, when merging, most likely will require an update on the suppresions.  Ping me beforehand, I can take care of the manual update.

**Comment by swegner** [comment]
This change looks good to me-- please rebase ontop of master to validate current suppressions and then we can merge.

I see @kennknowles is also marked for review; did you want a second review before merging?

**Comment by kennknowles** [comment]
I self-requested a review to make sure this showed up in https://github.com/pulls/review-requested, just as a system to not let reviews languish. LGTM from any committer is enough for me here.

**Comment by swegner** [comment]
Thanks, will merge when green. @HuangLED looks like something broke with the rebase.

**Comment by HuangLED** [comment]
Run Java PreCommit

**Comment by HuangLED** [comment]
Run Seed Job

**Comment by HuangLED** [comment]
Run Java PreCommit

**Comment by HuangLED** [comment]
synced to HEAD and now shows all green

**Pull request merged**